### ISSUE TITLE
[ Wayland ] Qutebrowser image upload issues only under Qtile

### ISSUE THREAD

**Issue by justinesmithies** [qtile/qtile#4434]
### Issue description

If I run the latest release of Qutebrowser using QT6 when tiled I can select to upload an image on Fosstodon but nothing happens after selecting the image. The same happens on Firefox too.
If I full screen either of them then i can select an image to upload and it uploads fine.

I tried the same in River and they both work perfectly as expected so it has to be something with Qtile.

### Version

0.22.2.dev364+g3f8a0008

### Backend

Wayland (experimental)

### Config

```python
https://codeberg.org/JustineSmithies/qtile-wayland-dotfiles
```

### Required

- [X] I have searched past issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by jwijenbergh** [comment]
images.google.com works fine for me on an upload with firefox. Any specific upload that is causing problems? I actually doubt this is a qtile issue but more like a portal/dbus/gtk/qt whatever issue that is not functioning correctly on your system

**Comment by justinesmithies** [comment]
It only happens when using fosstodon and I try to upload an image. I get the selection box and I select a file and nothing is uploaded if the window is tiled. If I full screen the same window then doing the same works . This issue happens on both Qutebrowser and Firefox.
If I do the same running under riverwm it all works as expected. So how can that not be a Qtile issue as nothing else has changed only the window manager ??

**Comment by jwijenbergh** [comment]
Right I missed the part where it only works in fullscreen, I dont have mastodon but I will have a look

**Comment by justinesmithies** [comment]
Yes all other sites seem to work.

**Comment by jwijenbergh** [comment]
![20230902_13h36m07s_grim](https://github.com/qtile/qtile/assets/46386452/a2ad70c1-6702-4e4a-b8c8-ffc2a1b5233f)

Is it in this screen with the clipboard icon? I could not reproduce (monadtall layout) https://mastodon.social/@testaqtilebug/110995335295675903

**Comment by jwijenbergh** [comment]
Ah! I can reproduce in qutebrowser

**Comment by justinesmithies** [comment]
Yes I click the paperclip icon then a file selection window opens then I click open and nothing happens unless Qutebrowser is full screened . The sane using Firefox too.

**Comment by jwijenbergh** [comment]
Interesting it works if I float the popup

**Comment by justinesmithies** [comment]
Happens all the time with me.

**Comment by jwijenbergh** [comment]
Can you reproduce it if you run qtile wayland inside of qtile wayland (so with a different modifier)?

**Comment by justinesmithies** [comment]
Well I did just test qtile within qtile and ran qutebrowser and the issue was still there ;)

**Comment by jwijenbergh** [comment]
Does river automatically float the file picker?

The issue disappears if I add this to my qtile config:

(inside float rules)

```python3
        Match(func=lambda c: c.is_transient_for()),
```
(this means automatically float a window if it has a parent, which is what I want anyways).

If I modify dwl to also not float popups/file pickers I also get this exact issue:

```diff
diff --git a/dwl.c b/dwl.c
index 4ff5c37..f09f770 100644
--- a/dwl.c
+++ b/dwl.c
@@ -1587,13 +1587,14 @@ mapnotify(struct wl_listener *listener, void *data)
 	 * we set the same tags and monitor than its parent, if not
 	 * try to apply rules for them */
 	 /* TODO: https://github.com/djpohly/dwl/pull/334#issuecomment-1330166324 */
-	if (c->type == XDGShell && (p = client_get_parent(c))) {
-		c->isfloating = 1;
-		wlr_scene_node_reparent(&c->scene->node, layers[LyrFloat]);
-		setmon(c, p->mon, p->tags);
-	} else {
-		applyrules(c);
-	}
+	//if (c->type == XDGShell && (p = client_get_parent(c))) {
+	//	c->isfloating = 1;
+	//	wlr_scene_node_reparent(&c->scene->node, layers[LyrFloat]);
+	//	setmon(c, p->mon, p->tags);
+	//} else {
+	//	applyrules(c);
+	//}
+	applyrules(c);
 	printstatus();
 
 unset_fullscreen:
```

**Comment by justinesmithies** [comment]
Yes i just tried river with no rules just a bare config and it floats the qutebrowser filepicker.
I've added `Match(func=lambda c: c.is_transient_for()),` to my config now and everything works as expected.
Thank you for your help and please feel free to close this issue if you're happy too.

**Comment by jwijenbergh** [comment]
Cool that it works. I will leave this open to remind myself to either document it or dig further on why it doesn't work without this rule

**Comment by Shinyzenith** [comment]
Why does tiling the file picker break the uploads? This feels like some underlying portal issue that needs to be solved as a pre-req.

**Comment by Shinyzenith** [comment]
Which portal did you use for the file upload? @justinesmithies was it the gtk portal?

**Comment by jwijenbergh** [comment]
> Why does tiling the file picker break the uploads? This feels like some underlying portal issue that needs to be solved as a pre-req.

I am feeling like this is not a qtile bug for sure

**Comment by justinesmithies** [comment]
I have xdg-desktop-portal-wlr installed

**Comment by Shinyzenith** [comment]
The wlr portal does not provide a file picker as far as I'm aware.

**Comment by uncomfyhalomacro** [comment]
Hello. For this issue, this is a new update from XDP. You will have to provide a portals.conf file at `/usr/share/xdg-desktop-portal` or at `~/.config/xdg-desktop-portal` e.g.

a file should be named after the `XDG_CURRENT_DESKTOP` or `XDG_SESSION_DESKTOP`. so for qtile, the file is named as `qtile-portals.conf` with contents:

```conf
[preferred]
default=wlr;gtk
```
